ZA2-Berenice Ogundele_JavaScript-Core2-Coursework-week1

BereniceOgundele opened the pull request:
After struggling with vs code, so happy that I finally had one mandatory test pass, yippee!!!

<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

amellie commented on mandatory/1-writers.js:
```diff
@@ -60,7 +60,16 @@ Exercise 1:
+
+// console.log("Hi, my name is ${writer.firstName} ${lastName}. I am ${writer.age} years old, and work as a ${writers.occupation}. =>  )
+ /*
 Exercise 2:
```
What about exercise 2 and 3?

amellie commented on mandatory/2-water-bottle.js:
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
+
+
 // Here is your starting point:
 let bottle = {
```
Have you run the test cases? A lot of them failed

amellie commented on mandatory/2-water-bottle.js:
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
     // calling this function should decrease your bottle volume by 10 units;
   },
   isFull: function () {
+    if (this.volume = 100) {
```
The syntax isn't correct. ``this.volume = 100`` literally sets the volume to 100, it does not check whether the volume is 100

amellie commented on mandatory/2-water-bottle.js:
```diff
@@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
     // this function should return true if your bottle is full;
   },
   isEmpty: function () {
+    if (this.volume = 0) {
```
same as above. Can you figure out how to fix this

amellie commented on mandatory/3-groceries.js:
```diff
@@ -30,15 +30,35 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
+for (day in weeklyMealPlan) {
```
excellent

amellie commented on mandatory/3-groceries.js:
```diff
@@ -30,15 +30,35 @@ Exercise 1:
+for (day in weeklyMealPlan) {
+  weeklyMealPlan[day].forEach((ingredient) => {
+    if (
+      !weekendGroceriesToBuy.includes(ingredient) &&
```
well done on checking whether the ingredient has been added or not

amellie reviewed (commented):
Well done -- It seems that you didn't include mandatory exercise #4 and #9?

Domenico-cyf commented:
Thank you so much Amalia! Will correct accordingly

On Sun, 11 Apr 2021, 23:44 Amalia, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Well done -- It seems that you didn't include mandatory exercise #4
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/4>
> and #9
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/9>
> ?
> ------------------------------
>
> In mandatory/2-water-bottle.js
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#discussion_r611254873>
> :
>
> > @@ -19,22 +19,42 @@ We made a start on this here by giving you the skeleton of our object.
>  You have to implement the missing features according to the specification.
>  */
>
> +
> +
>  // Here is your starting point:
>  let bottle = {
>
> Have you run the test cases? A lot of them failed
> ------------------------------
>
> In mandatory/2-water-bottle.js
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#discussion_r611255292>
> :
>
> >      // calling this function should decrease your bottle volume by 10 units;
>    },
>    isFull: function () {
> +    if (this.volume = 100) {
>
> The syntax isn't correct. this.volume = 100 literally sets the volume to
> 100, it does not check whether the volume is 100
> ------------------------------
>
> In mandatory/2-water-bottle.js
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#discussion_r611255317>
> :
>
> >      // this function should return true if your bottle is full;
>    },
>    isEmpty: function () {
> +    if (this.volume = 0) {
>
> same as above. Can you figure out how to fix this
> ------------------------------
>
> In mandatory/3-groceries.js
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#discussion_r611255511>
> :
>
> > @@ -30,15 +30,35 @@ Exercise 1:
>  */
>  // Gather all week item names into this array
>  let weeklyGroceriesToBuy = [];
> +for (day in weeklyMealPlan) {
>
> excellent
> ------------------------------
>
> In mandatory/3-groceries.js
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#discussion_r611255740>
> :
>
> >
> +let weekendGroceriesToBuy = [];
> +for (day in weeklyMealPlan) {
> +  weeklyMealPlan[day].forEach((ingredient) => {
> +    if (
> +      !weekendGroceriesToBuy.includes(ingredient) &&
>
> well done on checking whether the ingredient has been added or not
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/JavaScript-Core-2-Coursework-Week1/pull/72#pullrequestreview-633060006>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AR4MVTYI5QMLHJMQFSKJGHDTIIQ5VANCNFSM4ZNVMYVA>
> .
>

11ma commented on mandatory/5-recipes.js:
```diff
@@ -24,4 +24,41 @@ You should write and log at least 5 recipes
+  for(let ingredient of singleRecipe.ingredients) {
+  console.log(ingredient);
+  }
+}
```
This is well done Domenico, good variable names too, makes it readable and easy to understand the for loops. 🔥

BereniceOgundele committed:
2-mandatory

After struggling with vs code, so happy that I finally had one mandatory test pass, yippee!!!
